tchannel-core/src/main/java/com/uber/tchannel/handlers/RequestRouter.java

```diff
@@ -87,37 +89,27 @@ protected void channelRead0(final ChannelHandlerContext ctx, final Request reque
 
         // There is nothing to do if the connection is already destroyed.
         if (!ctx.channel().isActive()) {
-            logger.warn("drop request when channel is inActive");
             request.release();
+            logger.warn("drop request when channel is inActive");
             return;
         }
 
-        final ArgScheme argScheme = ArgScheme.toScheme(
-                request.getTransportHeaders().get(TransportHeaders.ARG_SCHEME_KEY)
-        );
-
-        if (argScheme == null) {
-            sendError(ErrorType.BadRequest,
-                "Expected incoming call to have \"as\" header set",
-                request, ctx);
+        if (request.getArgScheme() == null) {
+            sendError(ErrorType.BadRequest, "Expected incoming call to have \"as\" header set", request, ctx);
```

Comment:
Expected incoming call to have the \"as\" header set - perhaps definite article??? not sure

Reply:
yeah, we could go over error messages if there is a need - in a separate diff, perhaps.